ui/admin/app/routes/scopes/scope/workers/worker/create-tags.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class ScopesScopeWorkersWorkerCreateTagsRoute extends Route {
+  // =services
+
+  @service confirm;
+  @service intl;
+  @service router;
+
+  constructor() {
+    super(...arguments);
+    /**
+     * If user attempts to navigate away from unsaved changes, the user is
+     * asked to confirm that they would like to discard the changes.  If the
+     * user chooses discard, changes apiTags are cleared and the
+     * transition is retried.  If the user cancels discard, the transition is
+     * aborted.
+     */
+    this.router.on('routeWillChange', async (transition) => {
```

Comment:
Is there a reason you need to use the `router` instead of `willTransition` when you're already in the route?

Reply:
As Hoobastank would say: "The reason is me".

Updating now.